specifyweb/frontend/js_src/lib/components/Merging/index.tsx

```diff
@@ -299,12 +307,16 @@ function MergeButton<SCHEMA extends AnySchema>({
     'warningDialog'
   );
 
+  const [loadingBar, setLoadingBar] = React.useState(false);
+
   return (
     <>
       {!saveBlocked ? (
         <>
           {noShowWarning ? (
-            <Submit.Blue form={formId}>{treeText.merge()}</Submit.Blue>
+            <Submit.Blue form={formId} onClick={() => setLoadingBar(true)}>
```

Comment:
this isn't safe

if you click on a submit button, the form's onSubmit is only triggered if there are no validation errors
where as submit button's on click is called in every case.

thus if someone clicks this button while there are validation errors, they will be stuck with an infinite loading bar

Reply:
why do you even call `<Status>` from this component?

calling it from `<Merging>` makes much more sense - that component knows when the merge process has started (`onMerge`) and has access to `mergeId`